lib/ansible/modules/cloud/amazon/efs.py

```diff
@@ -335,19 +358,30 @@ def get_mount_targets_in_state(self, file_system_id, states=None):
 
         return list(targets)
 
-    def create_file_system(self, name, performance_mode):
+    def create_file_system(self, name, performance_mode, encrypt, kms_key_id):
         """
          Creates new filesystem with selected name
         """
         changed = False
         state = self.get_file_system_state(name)
+        params = {}
+        params['CreationToken'] = name
+        params['PerformanceMode'] = performance_mode
+        if encrypt:
+            params['Encrypted'] = encrypt
+        if kms_key_id is not None:
+            params['KmsKeyId'] = kms_key_id
+
         if state in [self.STATE_DELETING, self.STATE_DELETED]:
             wait_for(
                 lambda: self.get_file_system_state(name),
                 self.STATE_DELETED
             )
-            self.connection.create_file_system(CreationToken=name, PerformanceMode=performance_mode)
-            changed = True
+            try:
+                self.connection.create_file_system(**params)
+                changed = True
+            except ClientError as e:
+                self.module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
```

Comment:
.message doesn't exist at all when running Python 3. The exception handling guidelines have also added BotoCoreError to exceptions that should be caught (which do not have a .response attribute). https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-1